chain/params.go

```diff
@@ -1,18 +1,54 @@
 package chain
 
 import (
+	"errors"
 	"math/big"
+	"sort"
+	"strconv"
 
 	"github.com/0xPolygon/polygon-edge/types"
 )
 
+var (
+	// ErrBurntContractAddressMissing is the error when a contract address is not provided
+	ErrBurntContractAddressMissing = errors.New("burnt contract address missing")
+)
+
 // Params are all the set of params for the chain
 type Params struct {
 	Forks          *Forks                 `json:"forks"`
 	ChainID        int                    `json:"chainID"`
 	Engine         map[string]interface{} `json:"engine"`
 	Whitelists     *Whitelists            `json:"whitelists,omitempty"`
 	BlockGasTarget uint64                 `json:"blockGasTarget"`
+
+	// Governance contract where the token will be sent to and burnt in london fork
+	BurntContract map[string]string `json:"burntContract"`
+}
+
+// CalculateBurntContract calculates burn contract address for the given block number
+func (p *Params) CalculateBurntContract(number uint64) (types.Address, error) {
+	keys := make([]string, 0, len(p.BurntContract))
+	for k := range p.BurntContract {
+		keys = append(keys, k)
+	}
+
+	if len(keys) == 0 {
+		return types.ZeroAddress, ErrBurntContractAddressMissing
+	}
+
+	sort.Strings(keys)
+
+	for i := 0; i < len(keys)-1; i++ {
+		valUint, _ := strconv.ParseUint(keys[i], 10, 64)
+		valUintNext, _ := strconv.ParseUint(keys[i+1], 10, 64)
```

Comment:
Is maybe better to use temp `prev` variable which is firstly initialized with `prev, _ := strconv.ParseUint(keys[0], 10, 64)` and then update that variable in loop in order to not convert twice?